confidant/routes/credentials.py

```diff
@@ -804,19 +819,17 @@ def update_credential(id):
                 'conflicts': conflicts
             }
             return jsonify(ret), 400
+
+        # If the credential pair passed in the update is different from the
+        # decrypted credential pair of the most recent revision, assume that
+        # this is a new credential pair and update last_rotation_date
+        if credential_pairs != _cred.decrypted_credential_pairs:
```

Comment:
It is possible to enter this condition if you just change the key name while leaving the token the same - we deliberately ignore this case